### PR TITLE
Fix tracing for publisher and controller

### PR DESCRIPTION
**Pull request by sayanh**
**Description**

Changes proposed in this pull request:

- Fix tracing for publisher and controller

**Related issue(s)**

https://github.com/kyma-project/kyma/issues/10957

### CODE REVIEW

File: components/event-publisher-proxy/pkg/handler/nats/handler.go [hunk 1]
```diff
@@ -156,6 +156,7 @@ func (h *Handler) publishCloudEvents(writer http.ResponseWriter, request *http.R
 	}
 
 	h.receive(ctx, event)
+
```
Comment (marcobebway) [inline components/event-publisher-proxy/pkg/handler/nats/handler.go:159]: (nit) unnecessary changes in the file.

File: components/event-publisher-proxy/pkg/legacy-events/legacy.go [hunk 2]
```diff
@@ -123,6 +125,9 @@ func (t Transformer) TransformLegacyRequestsToCE(writer http.ResponseWriter, req
 		return nil
 	}
 
+	// Add tracing context to cloud events
+	tracing.AddTracingContextToCEExtensions(request.Header, event)
```
Comment (marcobebway) [inline components/event-publisher-proxy/pkg/legacy-events/legacy.go:129]: Since this is for legacy-events, Is it also handled for cloudevents?

Comment (sayanh) [reply components/event-publisher-proxy/pkg/legacy-events/legacy.go:129]: No, it's not. Fixed it.

File: components/event-publisher-proxy/pkg/handler/nats/handler.go (hunk 1 above)
Comment (sayanh) [reply components/event-publisher-proxy/pkg/handler/nats/handler.go:161]: I added intentionally to improve readability

File: components/eventing-controller/pkg/handlers/nats_test.go [hunk 3]
```diff
@@ -340,9 +340,9 @@ func TestIsValidSubscription(t *testing.T) {
 
 }
 
-func checkIsValid(sub *nats.Subscription, t *testing.T) error {
-	return checkValidity(sub, true, t)
-}
+//func checkIsValid(sub *nats.Subscription, t *testing.T) error {
+//	return checkValidity(sub, true, t)
+//}
```
Comment (marcobebway) [inline components/eventing-controller/pkg/handlers/nats_test.go:345]: If code is not used, can you please delete it?

Comment (sayanh) [reply components/eventing-controller/pkg/handlers/nats_test.go]: Fixed it

**Review by marcobebway — APPROVED** [review]
